BlockchainSdk/Common/BlockchainSdkConfig.swift

```diff
@@ -110,4 +110,50 @@ public extension BlockchainSdkConfig {
             self.mainnetApiKey = mainnetApiKey
         }
     }
+    
+    struct GetBlockCredentials {
+        let credentials: [Credential]
+        
+        public init(credentials: [Credential]) {
+            self.credentials = credentials
+        }
+    }
+}
+
+public extension BlockchainSdkConfig.GetBlockCredentials {
+    struct Credential {
+        let blockchain: Blockchain
+        let type: TypeValue
+        let value: String
+        
+        public init(blockchain: Blockchain, type: TypeValue, key: String) {
+            self.blockchain = blockchain
+            self.type = type
+            self.value = key
+        }
+    }
+    
+    enum TypeValue {
+        case blockBook
+        case rest
+        case jsonRpc
+        case rosseta
+    }
+}
+
+extension BlockchainSdkConfig.GetBlockCredentials {
+    func credential(for blockchain: Blockchain, type: TypeValue) -> String {
+        let credential = credentials.first { $0.blockchain.codingKey == blockchain.codingKey && $0.type == type }
+        return credential?.value ?? ""
+    }
+    
+    func credentials(type: TypeValue) -> [Blockchain: String] {
+        var resultJsonRpcApiKeys = [Blockchain: String]()
+        
+        credentials
+            .filter { $0.type == type }
+            .forEach { resultJsonRpcApiKeys[$0.blockchain] = $0.value }
+        
+        return resultJsonRpcApiKeys
+    }
```

Comment:
Можно через reduce собрать 

```suggestion
    func credentials(type: TypeValue) -> [Blockchain: String] {
        credentials
            .filter { $0.type == type }
            .reduce(into: [:]) { $0[$1.blockchain] = $1.value }
    }
```

Reply:
> credentials
>             .filter { $0.type == type }
>             .reduce(into: [:]) { $0[$1.blockchain] = $1.value }

Ага, спс, так получше будет